expirationd.lua

```diff
@@ -215,14 +206,18 @@ local Task_methods = {
     stop = function (self)
         if (get_fiber_id(self.guardian_fiber) ~= 0) then
             self.guardian_fiber:cancel()
-            while self.guardian_fiber:status() ~= 'dead' do
+            while self.guardian_fiber:status() ~= "dead" do
                 fiber.sleep(0.01)
             end
             self.guardian_fiber = nil
         end
         if (get_fiber_id(self.worker_fiber) ~= 0) then
-            self.worker_fiber:cancel()
-            while self.worker_fiber:status() ~= 'dead' do
+            if self.atomic_iteration then
+                self.worker_canceled = true
```

Comment:
I suggest to do it even atomic_iteration is force

Reply:
Probably you meant false, I left it as it was, because it is faster, and for the transaction it is more important that it completes correctly. Maybe it needs to be discussed